casper/src/main/scala/coop/rchain/casper/util/BondingUtil.scala

```diff
@@ -0,0 +1,164 @@
+package coop.rchain.casper.util
+
+import cats.effect.{Resource, Sync}
+import cats.implicits._
+
+import coop.rchain.casper.util.rholang.RuntimeManager
+import coop.rchain.casper.util.rholang.InterpreterUtil.mkTerm
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.{Blake2b256, Keccak256}
+import coop.rchain.crypto.signatures.{Ed25519, Secp256k1}
+import coop.rchain.rholang.interpreter.Runtime
+import coop.rchain.shared.PathOps.RichPath
+
+import java.io.PrintWriter
+import java.nio.file.{Files, Path}
+
+import monix.execution.Scheduler
+
+object BondingUtil {
+  def bondingForwarderAddress(ethAddress: String): String = s"${ethAddress}_bondingForwarder"
+  def bondingStatusOut(ethAddress: String): String        = s"${ethAddress}_bondingOut"
+  def transferStatusOut(ethAddress: String): String       = s"${ethAddress}_transferOut"
+
+  def bondingForwarderDeploy(bondKey: String, ethAddress: String): String =
+    s"""for(@purse <- @"${bondingForwarderAddress(ethAddress)}"; @pos <- @"proofOfStake"){
+       |  @(pos, "bond")!("$bondKey".hexToBytes(), "ed25519Verify", purse, "$ethAddress", "${bondingStatusOut(
+         ethAddress
+       )}")
+       |}""".stripMargin
+
+  def unlockDeploy[F[_]: Sync](ethAddress: String, pubKey: String, secKey: String)(
+      implicit runtimeManager: RuntimeManager,
+      scheduler: Scheduler
+  ): F[String] =
+    preWalletUnlockDeploy(ethAddress, pubKey, Base16.decode(secKey), s"${ethAddress}_unlockOut")
+
+  def bondDeploy[F[_]: Sync](amount: Long, ethAddress: String, pubKey: String, secKey: String)(
+      implicit runtimeManager: RuntimeManager,
+      scheduler: Scheduler
+  ): F[String] =
+    walletTransferDeploy(
+      0, //nonce
+      amount,
+      bondingForwarderAddress(ethAddress),
+      transferStatusOut(ethAddress),
+      pubKey,
+      Base16.decode(secKey)
+    )
+
+  def preWalletUnlockDeploy[F[_]: Sync](
+      ethAddress: String,
+      pubKey: String,
+      secKey: Array[Byte],
+      statusOut: String
+  )(implicit runtimeManager: RuntimeManager, scheduler: Scheduler): F[String] = {
+    require(Base16.encode(Keccak256.hash(Base16.decode(pubKey)).drop(12)) == ethAddress.drop(2))
+    val unlockSigDataTerm =
+      mkTerm(s""" @"__SCALA__"!(["$pubKey", "$statusOut"].toByteArray())""").right.get
+    for {
+      sigBytes <- Sync[F].delay {
+                   runtimeManager
+                     .captureResults(runtimeManager.emptyStateHash, unlockSigDataTerm)
+                     .head
+                     .exprs
+                     .head
+                     .getGByteArray
+                     .toByteArray
+                 }
+      unlockSigData = Keccak256.hash(sigBytes)
+      unlockSig     = Secp256k1.sign(unlockSigData, secKey)
+      _             = assert(Secp256k1.verify(unlockSigData, unlockSig, Base16.decode("04" + pubKey)))
```

Comment:
Can you extract out this "04" and add a descriptive label to it?

Reply:
I don't actually know what the `04` is. I just know that all public keys need to start with `04` in the `Secp256k1` library we use.

Reply:
It sounds like we should have at least someone research into why 04 is needed. I know we have this issue in which keys generated in Python for some reason don't work with the node. Maybe it is related?

Reply:
Yeah probably is related. I bet they would work if you gave them that extra byte